lib/buffer.js

```diff
@@ -462,8 +462,11 @@ Buffer.prototype.toString = function() {
   } else {
     result = slowToString.apply(this, arguments);
   }
-  if (result === undefined)
+  if (result === undefined) {
+    if (this.length > binding.kStringMaxLength)
+      throw new Error('Size exceeds max string length');
```

Comment:
Ok, didn't think so and I googled across this : 

Changing the text of an error message

> * Changing the text of an error message
> Not only should users not rely on these text messages, but they change anyways based on culture
> 

https://github.com/khellang/corefx-wiki/blob/master/Breaking-Change-Rules.md

Ok I stand by so.